worlds/pokemon_emerald/options.py

```diff
@@ -251,25 +363,61 @@ class RandomizeTrainerParties(Choice):
     option_completely_random = 4
 
 
-class AllowTrainerLegendaries(DefaultOnToggle):
+class TrainerPartyBlacklist(OptionSet):
+    """
+    Prevents listed species from appearing in opponent trainers' parties if opponent parties are randomized.
+
+    May be overridden if enforcing other restrictions in combination with this blacklist is impossible.
+
+    Use "_Legendaries" as a shortcut for legendary pokemon.
+    """
+    display_name = "Trainer Party Blacklist"
+    valid_keys = frozenset(species.label for species in data.species.values()) | {"_Legendaries"}
+
+
+class ForceFullyEvolved(Range):
     """
-    Enemy trainer pokemon can be replaced by legendaries. Only applied if Randomize Trainer Parties is not Vanilla.
+    Enemy trainers will only use fully evolved pokemon starting at the specified level.
```

Comment:
I think this sentence could be clarified slightly. Maybe something like "Enemy trainers' pokemon of the specified level or higher will automatically become fully evolved." Might be fine how it is, I can just see a few ways it could be misinterpreted

Reply:
Reworded the whole thing in [46f7c86](https://github.com/ArchipelagoMW/Archipelago/pull/2918/commits/46f7c86c42d9c9bb8da81295766fe84e202f2fbf)